packages/@react-aria/collections/src/BaseCollection.ts

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {Collection as ICollection, Key, Node} from '@react-types/shared';
+import {ReactElement, ReactNode} from 'react';
+
+export type Mutable<T> = {
+  -readonly[P in keyof T]: T[P]
+}
+
+/** An immutable object representing a Node in a Collection. */
+export class NodeValue<T> implements Node<T> {
```

Comment:
Is this the best name for it? `CollectionNode` maybe?

I didn't even really want to export this class (in favor of just using the `Node` interface) but it seems subclasses like `TableCollection` use it (and `Tree` uses it too for some reason).